File: devtools/client/shared/components/throttling/reducer.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  CHANGE_NETWORK_THROTTLING,
} = require("resource://devtools/client/shared/components/throttling/actions.js");

const INITIAL_STATE = {
  enabled: false,
  profile: "",
};

function throttlingReducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case CHANGE_NETWORK_THROTTLING: {
      return {
        enabled: action.enabled,
        profile: action.profile,
      };
    }
    default:
      return state;
  }
}

module.exports = throttlingReducer;

File: mobile/android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/Reducer.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.components.lib.state

/**
 * Reducers specify how the application's [State] changes in response to [Action]s sent to the [Store].
 *
 * Remember that actions only describe what happened, but don't describe how the application's state changes.
 * Reducers will commonly consist of a `when` statement returning different copies of the [State].
 */
typealias Reducer<S, A> = (S, A) -> S
